packages/core/src/Response.ts

```diff
@@ -1,16 +1,27 @@
 import type { NodeWithScore } from "./Node.js";
+import type { ChatMessage, ChatResponse } from "./llm/types.js";
 
-/**
- * Response is the output of a LLM
- */
-export class Response {
+export class Response implements ChatResponse {
```

Comment:
could we don't naming `Response` it's already a Web API

Reply:
how about `EngineResponse` (see new code)?